src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/SafeEvpPKeyHandle.OpenSsl.cs

```diff
@@ -177,5 +193,57 @@ public static SafeEvpPKeyHandle OpenPublicKeyFromEngine(string engineName, strin
 
             return Interop.Crypto.LoadPublicKeyFromEngine(engineName, keyId);
         }
+
+        /// <summary>
+        ///   Open a named public key using a named <c>OSSL_PROVIDER</c>.
+        /// </summary>
+        /// <param name="providerName">
+        ///   The name of the <code>provider</code> to process the key open request.
+        /// </param>
+        /// <param name="keyUri">
+        ///   The provider-assigned URI of the key to open.
+        /// </param>
+        /// <returns>
+        ///   The opened key.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        ///   <paramref name="providerName"/> or <paramref name="keyUri"/> is <see langword="null" />.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        ///   <paramref name="providerName"/> or <paramref name="keyUri"/> is the empty string.
+        /// </exception>
+        /// <exception cref="CryptographicException">
+        ///   the key could not be opened via the specified provider.
+        /// </exception>
+        /// <remarks>
+        ///   <para>
+        ///     Both <paramref name="providerName" /> and <paramref name="keyUri" /> must be trusted inputs.
+        ///   </para>
+        ///   <para>
+        ///     This operation will fail if OpenSSL cannot successfully load the named <code>provider</code>,
+        ///     or if the named <code>provider</code> cannot load the named key.
+        ///   </para>
+        ///   <para>
+        ///     The syntax for <paramref name="keyUri"/> is determined by each individual
+        ///     <code>provider</code>.
+        ///   </para>
+        /// </remarks>
```

Comment:
You changed the first `<code>provider</code>` to `<c>OSSL_PROVIDER</c>`, but left the others as they were.

Reply:
updated